samples/snippets/test_sample_delete_protection.py

```diff
@@ -0,0 +1,49 @@
+#  Copyright 2022 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import uuid
+
+import google.auth
+import pytest
+
+from quickstart import delete_instance
+from sample_delete_protection import create_instance
+from sample_delete_protection import get_delete_protection
+from sample_delete_protection import set_delete_protection
+
+PROJECT = google.auth.default()[1]
+INSTANCE_ZONE = "europe-central2-a"
+
+
+@pytest.fixture
+def autodelete_instance_name():
+    instance_name = "test-instance-" + uuid.uuid4().hex[:10]
+
+    yield instance_name
+
+    delete_instance(PROJECT, INSTANCE_ZONE, instance_name)
```

Comment:
Will these delete an instance even if has delete protection on? I'm curious about what happens if the test fails for some reason before line 45.

Reply:
When you try to delete a protected instance with this, you get an exception:

```
google.api_core.exceptions.BadRequest: 400 DELETE https://compute.googleapis.com:443/compute/v1/projects/diregapic-mestiv/zones/europe-central2-a/instances/protected: Invalid resource usage: 'Resource cannot be deleted if it's protected against deletion.'.
```

I'll add a check and unset the protect bit to make sure it's getting deleted by the end of the test.